servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/OpenTelemetryHttpServerFilter.java

```diff
@@ -135,9 +135,16 @@ private Single<StreamingHttpResponse> trackRequest(final StreamingHttpService de
         }
         final Context context = instrumenter.start(parentContext, request);
         try (Scope unused = context.makeCurrent()) {
-            final ScopeTracker tracker = new ScopeTracker(context, request, instrumenter);
+            System.err.println("Initiated context: " + Span.current().getSpanContext());
+            final ScopeTrackerV2 tracker = new ScopeTrackerV2(context, request, instrumenter);
             try {
                 Single<StreamingHttpResponse> response = delegate.handle(ctx, request, responseFactory);
+                // We attach this before the response resolves, because this has give the request time to
+                // traverse the entire pipeline. That means our transformation will happen as the last layer
+                // TODO: this is a concurrency bug: we've given away our request but are now modifying it 'later'
+                //  which could happen concurrently with some other modifications.
+                response = response.beforeFinally(() -> request.transformMessageBody(body ->
+                        transformBody(body, context).afterFinally(tracker::requestComplete)));
```

Comment:
If we want to capture the request events that are sourced from within the NettyHttpServer dispatch loop, we have to somehow set the context there as well, and order matters.